tasks/main.yml

```diff
@@ -66,3 +66,35 @@
   changed_when: sonar_version | string not in api_check_version.content
   notify: fail if expected sonar version not confirmed
   ignore_errors: "{{ ansible_check_mode }}"
+
+- name: "Ensure old admin password match"
+  uri:
+    method: POST
+    url: "http://{{ sonar_web_host }}:{{ sonar_web_port }}{{ sonar_web_context }}/api/authentication/login"
+    force_basic_auth: true
+    url_username: "{{ sonar_admin_user }}"
+    url_password: "{{ sonar_admin_old_password }}"
+    body: "login={{ sonar_admin_user }}&password={{ sonar_admin_old_password }}"
+    body_format: form-urlencoded
+    validate_certs: false
+    status_code:
+      - 200
+      - 401
+  register: login_check
+  when:
+    - sonar_admin_password_change
+
+- name: "Change old admin password new one"
+  uri:
+    method: POST
+    url: "http://{{ sonar_web_host }}:{{ sonar_web_port }}{{ sonar_web_context }}/api/users/change_password"
+    force_basic_auth: true
+    url_username: "{{ sonar_admin_user }}"
+    url_password: "{{ sonar_admin_old_password }}"
+    body: "login={{ sonar_admin_user }}&password={{ sonar_admin_password }}&previousPassword={{ sonar_admin_old_password }}"
```

Comment:
As a general rule of thumb, I would disable logging when credentials might leak to execution logs.

Reply:
What logs?  
url_* parameters is always no_log variable so in verbose output you can see only like this
```
"url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
"url_username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
```
Next,  `"body_format"` is specified like  `"form-urlencoded"` so  `body` will display as `login=********&password=********`

Reply:
Ok for me, i haven't check the `uri` documentation for a long time and wasn't sure.